File: Partition Array According to Given Pivot.py
class Solution(object):
    def pivotArray(self, nums, pivot):
        """
        :type nums: List[int]
        :type pivot: int
        :rtype: List[int]
        """
        size = len(nums)
        left_index = 0
        right_index = size-1
        result = list(range(0,size))
        for a in nums:
            if a < pivot:
                result[left_index] = a
                left_index += 1
            elif a > pivot:
                result[right_index] = a
                right_index -= 1
        
        while left_index <= right_index:
            result[left_index] = pivot
            left_index += 1

        left_index = right_index + 1
        right_index = size-1

        while left_index < right_index:
            result[left_index], result[right_index] = result[right_index], result[left_index]
            left_index += 1
            right_index -= 1
        
        return result

        

File: Partition Array According to Given Pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        queue<int> smaller;
        queue<int> equal;
        queue<int> larger;

        vector<int> result;

        for(auto &a: nums){
            if(a==pivot){
                equal.push(a);
            }
            else if(a<pivot){
                smaller.push(a);
            }
            else{
                larger.push(a);
            }
        }

        while(!smaller.empty()){
            result.push_back(smaller.front());
            smaller.pop();
        }

        while(!equal.empty()){
            result.push_back(equal.front());
            equal.pop();
        }

        while(!larger.empty()){
            result.push_back(larger.front());
            larger.pop();
        }

        return result;
    }
};
